src/macaron/__main__.py

```diff
@@ -74,15 +75,20 @@ def analyze_slsa_levels_single(analyzer_single_args: argparse.Namespace) -> None
     sys.exit(status_code)
 
 
-def verify_prov(verify_args: argparse.Namespace) -> None:
+def verify_prov(verify_args: argparse.Namespace) -> NoReturn:
```

Comment:
According to Python 3.11's documentation, `Never` is recommended to be used here instead - https://docs.python.org/3/library/typing.html#typing.NoReturn. I wonder if pyupgrade could detect such details :thinking:

Reply:
> According to Python 3.11's documentation, `Never` is recommended to be used here instead - https://docs.python.org/3/library/typing.html#typing.NoReturn (from Python 3.11). I wonder if pyupgrade could detect such details thinking

We have [ticket](https://github.com/oracle-samples/macaron/issues/52) to set `pyupgrade`'s version to `py 3.11. I'm expecting a lot of code rewrite when that Python version changes :wink: